graylog2-web-interface/src/views/components/widgets/SearchQueryExecutionInfoHelper.tsx

```diff
@@ -0,0 +1,114 @@
+import type { MessageResult, SearchTypeResult, SearchTypeResultTypes } from 'views/types';
+import type { SearchTypeIds } from 'views/logic/views/types';
+import OverlayDropdown from 'components/common/OverlayDropdown';
+
+type Props = {
+  currentWidgetMapping: SearchTypeIds,
+};
+
+const PopoverContainer = styled.div`
+  max-width: 500px;
+  padding: 5px;
+`;
+
+const QueryHelpButton = styled(Button)`
+  padding: 6px 8px;
+`;
+
+type WidgetExecutionData = {
+  total: number,
+  duration: number,
+  timestamp: string,
+  effectiveTimerange: SearchTypeResult['effective_timerange'] | MessageResult['effectiveTimerange']
+}
+const HelpPopover = ({ widgetExecutionData }: { widgetExecutionData: WidgetExecutionData}) => (
+  <PopoverContainer>
+    <p><strong>Execution Info</strong></p>
+    <Table condensed>
+      <tbody>
+        <tr>
+          <td><i>Executed at:</i></td>
+          <td aria-label="Executed at"><Timestamp dateTime={widgetExecutionData?.timestamp} /></td>
+        </tr>
+        <tr>
+          <td><i>Executed in:</i> </td>
+          <td>{numeral(widgetExecutionData?.duration).format('0,0')}ms</td>
+        </tr>
+        <tr>
+          <td colSpan={2}><i>Effective time range:</i></td>
+        </tr>
+        <tr>
+          <td>From</td>
+          <td aria-label="Effective time range from"><Timestamp dateTime={widgetExecutionData?.effectiveTimerange?.from} format="complete" /></td>
+        </tr>
+        <tr>
+          <td>To</td>
+          <td aria-label="Effective time range to"><Timestamp dateTime={widgetExecutionData?.effectiveTimerange?.to} format="complete" /></td>
+        </tr>
+        <tr>
+          <td><i>Total results:</i></td>
+          <td>{numeral(widgetExecutionData?.total).format('0,0')}</td>
+        </tr>
+      </tbody>
+    </Table>
+  </PopoverContainer>
+);
+
+const SearchQueryExecutionInfoHelper = ({ currentWidgetMapping }: Props) => {
+  const [open, setOpen] = useState(false);
+  const result = useAppSelector(selectCurrentQueryResults);
+  const currentWidgetSearchType = useMemo<SearchTypeResultTypes[keyof SearchTypeResultTypes]>(() => {
+    const searchTypeId = currentWidgetMapping?.toJS()?.[0];
+
+    return result?.searchTypes?.[searchTypeId];
+  }, [currentWidgetMapping, result?.searchTypes]);
+
+  const widgetExecutionData = useMemo<WidgetExecutionData>(() => ({
+    effectiveTimerange: (currentWidgetSearchType as MessageResult)?.effectiveTimerange || (currentWidgetSearchType as SearchTypeResult)?.effective_timerange,
+    total: currentWidgetSearchType?.total,
+    duration: result?.duration,
+    timestamp: result?.timestamp,
+
+  }), [currentWidgetSearchType, result?.duration, result?.timestamp]);
+
+  const onMenuToggle = useCallback(() => {
+    setOpen((cur) => !cur);
+  }, []);
+
+  return (
+    <OverlayDropdown show={open}
+                     toggleChild={<QueryHelpButton bsStyle="link"><Icon name="question-circle" /></QueryHelpButton>}
```

Comment:
It would be nice to use a button with a smaller height here, so the widget content receives more space.

Before
![image](https://github.com/Graylog2/graylog2-server/assets/46300478/4f0ff18a-f570-409b-bb98-8fd0eeece58b)

After:
![image](https://github.com/Graylog2/graylog2-server/assets/46300478/c74809ac-3546-4b45-baf4-f406b809bef2)

Reply:
I can imagine something like this
![image](https://github.com/Graylog2/graylog2-server/assets/46300478/ef3ff9f6-4613-4149-95cd-96dba2dfde7e)

It could work well in combination with making the time range (5 minutes ago) + the icon the trigger to open the popover.

Reply:
i also like to display the icon the gray we use for the widget actions, to put the focus on the widget content.